Add package for pep8-naming

dopplershift opened the pull request:
It's a flake8 plugin to enforce pep8 "rules" for naming. I hate blind enforcement, but it's been nice having consistency from the outset.

I *think* this gives me a complete dev environment using just defaults + conda-forge.

pelson commented:
I think you need to put the recipe in the "recipes" folder. (note that **is** now the recipes folder, not the "ready_recipes" folder)

pelson commented on pep8-naming/meta.yaml:
```diff
@@ -0,0 +1,36 @@
+
+  run:
+    - python
+    - setuptools
```
Is this really a runtime dependency? Is it because it uses entrypoints?

dopplershift replied:
Well, it's not importing setuptools anywhere....so maybe not?

pelson commented:
> It's a flake8 plugin to enforce pep8 "rules" for naming

Interesting. Is there a reason that it doesn't have pep8 as a dependency? Are there any CLI scripts installed by this package which can be tested with

```
test:
   commands:
      - whatever
```

dopplershift commented:
It's a plugin to flake8. I supposed I should maybe add flake8 to the run requirements. I'm not sure if running flake8 itself would be useful to ensure the plugin package was built correctly.

dopplershift commented:
Ok:
- moved the files
- replaced setuptools runtime dependency with flake8
- changed test to see if flake8 finds the plugin

I ended up needing to rebase on master, so I just went ahead and squashed everything.

dopplershift commented:
Looks like something isn't quite right with smithy on Travis (ok, I just wanted to try out highlighted tracebacks):
```pytb
Traceback (most recent call last):
  File "/Users/travis/miniconda/bin/conda-smithy", line 9, in <module>
    load_entry_point('conda-smithy==0.4.2', 'console_scripts', 'conda-smithy')()
  File "/Users/travis/miniconda/lib/python2.7/site-packages/conda_smithy/cli.py", line 154, in main
    args.subcommand_func(args)
  File "/Users/travis/miniconda/lib/python2.7/site-packages/conda_smithy/cli.py", line 119, in __call__
    ci_register.add_project_to_travis(owner, repo)
  File "/Users/travis/miniconda/lib/python2.7/site-packages/conda_smithy/ci_register.py", line 117, in add_project_to_travis
    with open(os.path.expanduser('~/.conda-smithy/github.token'), 'r') as fh:
IOError: [Errno 2] No such file or directory: '/Users/travis/.conda-smithy/github.token'
Traceback (most recent call last):
  File ".CI/create_feedstocks.py", line 122, in <module>
    subprocess.check_call(['conda', 'smithy', 'register-feedstock-ci', feedstock_dir] + owner_info)
  File "/Users/travis/miniconda/lib/python2.7/subprocess.py", line 540, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['conda', 'smithy', 'register-feedstock-ci', '/var/folders/gw/_2jq29095y7b__wtby9dg_5h0000gn/T/tmp729Tov__feedstocks/pep8-naming-feedstock', '--organization', 'conda-forge']' returned non-zero exit status 1
```

pelson commented on recipes/pep8-naming/meta.yaml:
```diff
@@ -0,0 +1,36 @@
+
+test:
+  commands:
+    - "flake8 --version | grep naming:"
```
We've lost the import test - that looked like a good thing to try too... was there a reason for dropping it?

dopplershift replied:
It seemed superfluous vs. having flake8 run and pick up the plugin (which will do the import).

I'll defer to your view here, though, if you think it serves a purpose.

pelson replied:
> I'll defer to your view here, though, if you think it serves a purpose.

The only purpose I can come up with is if your pep8 check changes or adds a "naming:" at some point which doesn't relate to this. It is pretty unlikely IMHO.

Question: Does grep just work on our AppVeyor builds? Seems it does! :+1:

pelson commented:
Sorry @dopplershift - fixed in #73.

Nothing to worry about in terms of your changes though.
